### PR TITLE
Update validation tests for 3D textures with BC compression

### PR DESCRIPTION
**Pull request by mehmetoguzderin**
Partially fixes https://github.com/gpuweb/cts/issues/3761

Please let me know if I missed or overlooked anything. Your guidance would be appreciated. Thank you!

<hr>

**Requirements for PR author:**

- [x] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [x] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [x] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)
- [x] Test have be tested with compatibility mode validation enabled and behave as expected. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [ ] Tests are properly located in the test tree.
- [ ] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [ ] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [ ] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

File: src/webgpu/api/validation/createTexture.spec.ts [hunk 1]
```diff
@@ -476,8 +475,10 @@ g.test('texture_size,default_value_and_smallest_size,compressed_format')
       })
   )
   .beforeAllSubcases(t => {
-    const { format } = t.params;
+    const { dimension, format } = t.params;
     const info = kTextureFormatInfo[format];
+    // Skip if the dimension is 3D and format is not compatible
+    t.skipIf(dimension === '3d' && !info.texture3D, `${format} is not compatible with 3D texture`);
```
Comment (kainino0x) [inline src/webgpu/api/validation/createTexture.spec.ts:481]: Skipping is intended for test cases that work on some systems and not others. Instead, filter this out of the "params" earlier, using `.filter()` or `.unless()`.

File: src/webgpu/api/validation/createTexture.spec.ts [hunk 2]
```diff
@@ -101,7 +101,7 @@ g.test('zero_size_and_usage')
 
 g.test('dimension_type_and_format_compatibility')
   .desc(
-    `Test every dimension type on every format. Note that compressed formats and depth/stencil formats are not valid for 1D/3D dimension types.`
+    `Test every dimension type on every format. Note that compressed formats are not valid for 1D dimension type with only partial support for 3D dimension type and depth/stencil formats are not valid for 1D/3D dimension types.`
```
Comment (kainino0x) [inline src/webgpu/api/validation/createTexture.spec.ts:104]: nit: simplify:
```suggestion
    `Test every dimension type on every format. Depth/stencil formats only support 2d;
    compressed formats support either 2d or 2d+3d.`
```

File: src/webgpu/api/validation/createTexture.spec.ts (hunk 1 above)
Comment (mehmetoguzderin) [reply src/webgpu/api/validation/createTexture.spec.ts]: @kainino0x I tried to address, PTAL! Thank you!

File: src/webgpu/api/validation/createTexture.spec.ts [hunk 3]
```diff
@@ -474,9 +474,13 @@ g.test('texture_size,default_value_and_smallest_size,compressed_format')
...
         ];
       })
+      // Filter if the dimension is 3D and format is not compatible
+      .filter(
+        ({ dimension, format }) => !(dimension === '3d' && !kTextureFormatInfo[format].texture3D)
+      )
   )
   .beforeAllSubcases(t => {
-    const { format } = t.params;
+    const { _dimension, format } = t.params;
```
Comment (kainino0x) [inline src/webgpu/api/validation/createTexture.spec.ts:483]:
```suggestion
    const { format } = t.params;
```

File: src/webgpu/api/validation/createTexture.spec.ts [hunk 4]
```diff
@@ -474,9 +474,13 @@ g.test('texture_size,default_value_and_smallest_size,compressed_format')
           { size: [blockWidth, blockHeight, 1], _success: true },
         ];
       })
+      // Filter if the dimension is 3D and format is not compatible
+      .filter(
+        ({ dimension, format }) => !(dimension === '3d' && !kTextureFormatInfo[format].texture3D)
+      )
```
Comment (kainino0x) [inline src/webgpu/api/validation/createTexture.spec.ts:480]: nit (IMO easier to read):

```suggestion
      .unless(
        ({ dimension, format }) => dimension === '3d' && !kTextureFormatInfo[format].texture3D
      )
```

**Review by kainino0x — APPROVED** [review]

**Comment by mehmetoguzderin** [comment]
@kainino0x thank you for the great suggestions!

**Pull request merged**